### PR TITLE
Added d2l-cpd dependency

### PR DESCRIPTION
**Pull request by BraydonKains**
This adds https://github.com/Brightspace/continuous-professional-development to BSI. Related PR: https://git.dev.d2l/projects/CORE/repos/lms/pull-requests/8093/overview

### CODE REVIEW

**Comment by BraydonKains** [comment]
Dave and Stacey, I added you to this PR because there was some stuff in the package-lock that looked suspicious to me. It's hard for me to tell if anything is particularly an issue though.

File: package-lock.json [hunk 1]
```diff
@@ -3868,6 +3891,51 @@
...
+        "d2l-menu": "github:BrightspaceUI/menu#semver:^2",
+        "d2l-navigation": "github:BrightspaceUI/navigation#semver:^3",
+        "d2l-resize-aware": "github:BrightspaceUI/resize-aware#semver:^1",
+        "d2l-table": "github:BrightspaceUI/table#semver:^2",
+        "d2l-tabs": "github:BrightspaceUI/tabs#semver:^1",
+        "dayjs": "^1.8.19",
+        "lit-element": "^2"
+      },
+      "dependencies": {
+        "d2l-navigation": {
```
Comment (svanherk) [inline package-lock.json]: I think this will be a problem, you'll need to move to `#semver:^4`

Comment (svanherk) [reply package-lock.json]: I _expect_ this will fail the build... if it doesn't we'll have to look into that 😛

Comment (jwong-d2l) [reply package-lock.json]: We'll have to test our repo with v4 navigation.

Comment (BraydonKains) [reply package-lock.json]: Okay good to know. This PR will be on hold then while we test out bumping that dependency on our end.

Comment (BraydonKains) [reply package-lock.json]: Do you mean it should fail `npm run build`? Because it doesn't seem to. 🤔

Comment (dlockhart) [reply package-lock.json]: It should fail the `test:package-lock` script.

Comment (dlockhart) [reply package-lock.json]: Which it did! 🎉

Comment (BraydonKains) [reply package-lock.json]: Oh okay good. We'll try moving to v4 then and come back to this!

File: .npmrc [hunk 2]
```diff
@@ -1 +0,0 @@
-engine-strict=true
```
Comment (dlockhart) [inline .npmrc]: This was previously need for... reasons. Why is it going away?

File: web-components/d2l-cpd-report.js [hunk 3]
```diff
@@ -0,0 +1 @@
+import 'd2l-cpd/src/components/cpd-record-report';
```
Comment (dlockhart) [inline web-components/d2l-cpd-report.js]: Probably want a `.js` extension on this just for consistency.

File: .npmrc (hunk 2 above)
Comment (BraydonKains) [reply .npmrc]: Wasn't removed on purpose, I don't know what does this. I think it has something to do with the unbundle script.

File: package-lock.json [hunk 4]
```diff
@@ -4511,7 +4558,7 @@
       }
     },
     "d2l-sequence-viewer": {
-      "version": "github:Brightspace/d2l-sequence-viewer#e7b61fe8f4f0d8bd8e9cd22b10fff538ba731c1a",
+      "version": "github:Brightspace/d2l-sequence-viewer#f499116f50896b6567f56182b090df15fa082aca",
```
Comment (dlockhart) [inline package-lock.json]: @ChrisKraljevicD2L FYI

File: package-lock.json [hunk 5]
```diff
@@ -4532,7 +4579,7 @@
       }
     },
     "d2l-sequences": {
-      "version": "github:BrightspaceHypermediaComponents/sequences#01f01f614cda338f906c37cc048bc729648b749c",
+      "version": "github:BrightspaceHypermediaComponents/sequences#cf455c740e9dc229049c52bb9f2b5647d1d3e347",
```
Comment (dlockhart) [inline package-lock.json]: @ChrisKraljevicD2L, @pbrown-d2l FYI

File: package-lock.json [hunk 6]
```diff
@@ -4673,6 +4720,12 @@
         "assert-plus": "^1.0.0"
       }
     },
+    "dayjs": {
+      "version": "1.8.20",
+      "resolved": "https://registry.npmjs.org/dayjs/-/dayjs-1.8.20.tgz",
+      "integrity": "sha512-mH0MCDxw6UCGJYxVN78h8ugWycZAO8thkj3bW6vApL5tS0hQplIDdAQcmbvl7n35H0AKdCJQaArTrIQw2xt4Qg==",
+      "dev": true
+    },
```
Comment (dlockhart) [inline package-lock.json]: What's this being used for again? I don't love bundling in all kinds of date/time stuff...

Comment (jwong-d2l) [reply package-lock.json]: some crazy utc timezoney stuff that we didn't want to write ourselves, and wasn't in the base library.

Comment (dlockhart) [reply package-lock.json]: Ok, so it might go away once things switch over to the new date/time inputs?

Comment (jwong-d2l) [reply package-lock.json]: I hope so.  We tried to pull it once and things got weird.  I have chatted with Jeff about our use case though.

Comment (dlockhart) [reply package-lock.json]: Ok well either way we'll make sure it handles your case. We've written some UTC->user local timezone conversion stuff that'll hopefully be useful.

**Pull request merged**